stats.py

```diff
@@ -0,0 +1,5 @@
+def count_types(categories):
```

Comment:
consider replacing this function with https://docs.python.org/3/library/collections.html#collections.Counter

Reply:
> A Counter is a dict subclass for counting hashable objects. It is a collection where elements are stored as dictionary keys and their counts are stored as dictionary values. Counts are allowed to be any integer value including zero or negative counts. The Counter class is similar to bags or multisets in other languages.